### DISCUSSION TITLE
What is the best way to search for a hash?

### DISCUSSION THREAD

**Discussion by TechoTiger** [ChatGPTNextWeb/NextChat#5929] [Q&A] [answered]
I got sha256 hashes (hexdigest) of the text chunks as a column in my table. To check if a text is allready in the table I need to find the hash of a probably new chunk.

I tried
`table.search().select(["sha"]).where("sha='xxx'").to_arrow()`
which takes about 250ms.

Is there a better way? Is it possible to build an index that speeds it up?

**Comment by legendm00n** [comment, marked as answer]
You can improve the performance of your query by creating an index on the sha column.
In fact, indexing can drastically speed up the search for specific values, as it allows the database to quickly locate rows based on the indexed column.
If you're using a SQL-based database like PostgreSQL, MySQL, or SQLite, you can create an index on the `sha `column to improve the query speed.
Here is the sample query for PostgreSQL.
```sql
CREATE INDEX sha_index ON your_table_name (sha);
```
This will ensure that your query to search for the `sha` value is much faster because the database can use the index to look up the value in a more optimized
